nemo/src/model/rule_model/term.rs

```diff
@@ -367,6 +375,58 @@ impl Term {
             Term::Function(sub) => sub.subterms.iter().any(Self::aggregate_subterm_recursive),
         }
     }
+
+    /// Evaluates a constant (numeric) term
+    /// We do this by casting everything to `f64` as it seems like the most general number type.
```

Comment:
It's really not, for example, `9007199254740992.0 + 1.0 == 9007199254740992.0` holds for `f64`, but surely not for `i64`.

Reply:
It turns out I had the type information there all along and its not too complicated to use the evaluation code from the physical layer for the evaluation here